kernel/src/snapshot.rs

```diff
@@ -109,6 +108,12 @@ pub struct Snapshot {
     schema: Schema,
 }
 
+impl Drop for Snapshot {
+    fn drop(&mut self) {
+        println!("Dropping snapshot");
+    }
+}
+
```

Comment:
nit: should this use tracing - i.e. `debug!`?

Reply:
I wasn't actually trying to push this in the first place -- but we could do `debug!` if you think that's helpful to have in general?

Reply:
I think tracing drops could be useful, but it's probably fine to remove for now and we can add this in if we're ever debugging a case where we think it's important.